core/org.wso2.carbon.ndatasource.rdbms/src/main/java/org/wso2/carbon/ndatasource/rdbms/CorrelationLogInterceptor.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.ndatasource.rdbms;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.tomcat.jdbc.pool.interceptor.AbstractQueryReport;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.sql.*;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Time-Logging interceptor for JDBC pool.
+ * Logs the time taken to execute the query in each pool-ed connection.
+ */
+
+public class CorrelationLogInterceptor extends AbstractQueryReport {
+
+    private static final Log correlationLog = LogFactory.getLog("CORRELATION_LOG");
+
+    private static final String CORRELATION_LOG_TIME_TAKEN_KEY = "delta";
+    private static final String CORRELATION_LOG_TIME_TAKEN_UNIT = " ms";
```

Comment:
Could you please explain the necessity of a white-space character prior to "ms"?

Reply:
It is to add a space next to time value printed like for example: 10 ms